tests/test_maker_utils.py

```diff
@@ -47,8 +47,6 @@ def test_no_extra_fields(node_class, manifest):
 
     schema_keys = set()
     subschemas = [schema]
-    if "allOf" in schema:
-        subschemas.extend(schema["allOf"])
```

Comment:
I could use help understanding what this does. Would you explain why this isn't needed? I'm thinking it would cover the case where a RAD schema extends an existing schema by doing something like:
```yaml
allOf:
  - $ref: other_schema-1.0.0.yaml
  - extra_stuff..
```
kind of like what is done in [file_date-1.0.0.yaml](https://github.com/spacetelescope/rad/blob/1c7dd7fee74ba7a8b57c6250ed4399e35199a7a8/src/rad/resources/schemas/tagged_scalars/file_date-1.0.0.yaml#L8-L9). I'm not seeing any schema for the nodes tested here that use that pattern. Is there something in the design of the members of NODE_CLASSES that prevents this kind of schema extension (and makes this part of the test unnecessary in addition to being unused)?

Reply:
Look at the parameterization: https://github.com/spacetelescope/roman_datamodels/blob/e9d93fdef8c80f26036bd09a0404d9ba89a157fd/tests/test_maker_utils.py#L38

The `FileDate` object generated by the `file_date-1.0.0.yaml` is NOT a `TaggedObjectNode` it is a `TaggedScalarNode`.

This test in not applicable to `TaggedScalarNode` objects because by construction they cannot have any `keys`. The same applies to `TaggedScalarNode` objects as well.

Reply:
In general, the `TaggedObjectNode` objects should not need to have a top-level `allOf` combiner.

Reply:
Thanks for the extra details.

I provided the link to file date to provide an example of the pattern. Thanks for the confirmation that as I mentioned the pattern is not currently used in rad schemas that turn into TaggedObjectNodes.

If I add a schema that uses the pattern to rad and a maker utility to roman_datamodels all tests pass without issue if the changes are based off of rad main and roman_datamodels main. If however I base the changes off of the source branch for this PR (with the lines discussed here removed) the roman_datamodels (but not rad) tests fail with:
```
AssertionError: Dummy instance has extra keys: ...
```
due to the `test_no_extra_fields` test failing to step into the allOf at the base of the added schema.

The pattern of extending one schema with an allOf and ref is used extensively in JWST schemas and there appears to be nothing in RAD that prevents this pattern and nothing in roman_datamodels that prevents turning an extended schema into functional TaggedObjectNode.

If this pattern should not be used in roman wouldn't it be preferable to either:
- add a restriction in the rad schema (if possible)
- add a test in rad (several appear to exist to further check schemas)

instead of removing these lines so that a schema that extends another schema fails the `test_no_extra_fields` test?

If so, those changes seem outside the scope of this PR.

Reply:
> The pattern of extending one schema with an allOf and ref is used extensively in JWST schemas and there appears to be nothing in RAD that prevents this pattern and nothing in roman_datamodels that prevents turning an extended schema into functional TaggedObjectNode.
> 
> If this pattern should not be used in roman wouldn't it be preferable to either:
> 
> * add a restriction in the rad schema (if possible)
> * add a test in rad (several appear to exist to further check schemas)
> 
> instead of removing these lines so that a schema that extends another schema fails the `test_no_extra_fields` test?
> 
> If so, those changes seem outside the scope of this PR.

As it stands, virtually all changes to the structure of a schema in RAD will naturally induce failures in `roman_datamodels` because currently there is not a solution to dynamically generating node objects filled with null data from the schemas. Effectively, this means any change such as you describe is expected to induce failures in `roman_datamodels`. Thus it is expected that all changes in RAD are developed alongside changes to `roman_datamodels`.

If there becomes a case where an `allOf` combiner is needed in a `TaggedObjectNode`'s schema then this can be re-introduced. However, there is not currently a use case for this and it actively discourages unnecessary usage of a top level `allOf` combiner. I have already had to clean up several unnecessary uses of the `allOf` combiner in RAD, for some examples see spacetelescope/rad#267. While these unnecessary uses of the `allOf` do not effect how the schema is applied, the do unnecessarily complicate the schemas, which makes it more difficult for new/external users to understand the structure of their datamodels (the structure is dynamically generated from these schemas meaning the only reference users have for the structure is the schemas, which is already a barrier to some users).

I would rather not have to continually notice and fix small issues like spacetelescope/rad#267, but it is involved to add such a test in RAD. This is a relatively simple thing to address here, and as already stated there is an implied expectation of co-developing most changes significant changes between RAD and `roman_datamodels` meaning the home of such checks does not matter.

All this being said I will revoke this change as there are clear objections to it.

Reply:
Thanks for the clarification. I don't have any objection to preventing the extension of existing schemas (but I don't also have as much involvement in this project as @ddavis-stsci @PaulHuwe @schlafly etc).

Thanks for helping me understand the impact of removing these lines of code. I don't think this change should be in a PR changing the coverage measurement to include tests.